### ISSUE TITLE
Mesa 24 rendering issue 

### ISSUE THREAD

**Issue by DrogonMar** [o3de/o3de#18088]
**Describe the bug**
On Mesa 24 the viewport for the editor and game are black but with some parts working like specular.
If you use an older mesa version like 23.3.3 you will get the expected result.

**Assets required**
Any project works. 

**Steps to reproduce**
1. Open the Editor or GameLauncher.
2. See that you can only see black.

**Expected behavior**
The world being rendered as expected.

**Actual behavior**
A black world with specular.

**Screenshots/Video**
![image](https://github.com/o3de/o3de/assets/17402379/a28239e6-3d75-4d96-9384-8353c4671329)
![image](https://github.com/o3de/o3de/assets/17402379/08ad1cdf-b77b-4fdf-991d-f8a1a2575ec8)

**Found in Branch**
O3DE 2310.3 ~~and in the stabilization version 24090~~

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
[1577b1e742358461d8b2e6cfafd937637315c9d5](https://github.com/o3de/o3de/commit/1577b1e742358461d8b2e6cfafd937637315c9d5)
~~[454b5625e55059955a0f35d84d62101e754bf7fd](https://github.com/o3de/o3de/commit/454b5625e55059955a0f35d84d62101e754bf7fd)~~

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Linux 
 - Version Fedora 40 (or a distro with mesa 24)
 - CPU AMD Ryzen 9 7950X
 - GPU AMD Radeon RX 6950 XT
 - Memory 64GB

**Additional context**
Other programs don't seem to have an issue with the change to mesa 24.

**Comment by Andre-LA** [comment]
This is also observed here on 23.10.3

On stabilization/2409, the `Editor` segfaults, the crash happens on `EditorViewportWidget::SetAsActiveViewport`, on a NULL de-reference of `viewportContextManager`:
https://github.com/o3de/o3de/blob/454b5625e55059955a0f35d84d62101e754bf7fd/Code/Editor/EditorViewportWidget.cpp#L2119-L2121

Using an older mesa also "fix" the problem on 24.09

**Found in Branch:**
`stabilization/2409`

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository:**
454b5625e55059955a0f35d84d62101e754bf7fd

**Desktop/Device:**
 - Device: PC
 - OS: Linux
 - Version: Fedora 40
 - CPU & GPU: AMD Ryzen 5 5600G
 - Memory: 16GB

**Comment by DrogonMar** [comment]
Ive switched my GPU for a Intel Arc 750 for this test, Ill switch back to my AMD gpu after this.
Enabling vulkan validation shows this:

```
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkQueueBindSparse-queuetype ] Object 0: handle = 0x563234e379d0, type = VK_OBJECT_TYPE_QUEUE; | MessageID = 0xc8253d41 | vkQueueBindSparse(): a non-memory-management capable queue -- VK_QUEUE_SPARSE_BINDING_BIT not set. The Vulkan spec states: The queue must support sparse binding operations (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkQueueBindSparse-queuetype)
```

This log isn't present in mesa 23 and for reference [heres my gpu](http://vulkan.gpuinfo.org/displayreport.php?id=31616#queuefamilies)

**Comment by DrogonMar** [comment]
Focusing on 24.09 now and it looks like the null de-reference is when mesa is trying to free when it enumerates the devices [here.](https://github.com/o3de/o3de/blob/stabilization/2409/Gems/Atom/RHI/Vulkan/Code/Source/RHI/PhysicalDevice.cpp#L44)

![image](https://github.com/o3de/o3de/assets/17402379/b2340974-e583-438a-9593-1b6b1e6f9da9)

**Comment by DrogonMar** [comment]
Commenting out the RHI_vkFreeFunction in VkAllocator.cpp does get the editor running and can confirm the black screen issue on 23.10 is not in 24.09 so this issue is now focused on just this weird dealloc issue with mesa 24.

**Comment by pollend** [comment]
is this change still relevant, i have a similar gpu. looks like none of the textures are loading i found it was because my gpu has a dedicated queue for bindless.

https://github.com/o3de/o3de/pull/17905

**Comment by DrogonMar** [comment]
> is this change still relevant, i have a similar gpu. looks like none of the textures are loading i found it was because my gpu has a dedicated queue for bindless. looks very similar.
> 
> #17905

If your using version 23.10 then its expected, its fixed in the newest stabilization branch 24.09 although with this version it now segfaults due to some weird issue with mesa 24 and o3d3's custom allocator.

**Comment by DrogonMar** [comment]
@pollend I really should've looked more into your PR, Andre-LA found out after a bunch of testing that your PR is the thing that fixes it of course, with that said this is now closed.
